src/primitive_shape.rs

```diff
@@ -0,0 +1,177 @@
+// Copyright 2024 the Kurbo Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#![allow(unused_qualifications)]
+
+use crate::{Point, Rect, Shape};
+use alloc::boxed::Box;
+
+mod _never_shape {
+    use super::*;
+    use crate::PathEl;
+    /// An uninhabited type that implements shape.
+    #[derive(Debug, Clone, Copy, PartialEq)]
+    #[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+    #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+    pub enum NeverShape {}
+    impl Shape for NeverShape {
+        type PathElementsIter<'a> = core::iter::Empty<PathEl>;
+        fn path_elements(&self, _: f64) -> Self::PathElementsIter<'_> {
+            unreachable!()
+        }
+        fn area(&self) -> f64 {
+            unreachable!()
+        }
+        fn perimeter(&self, _: f64) -> f64 {
+            unreachable!()
+        }
+        fn winding(&self, _: Point) -> i32 {
+            unreachable!()
+        }
+        fn bounding_box(&self) -> Rect {
+            unreachable!()
+        }
+    }
+}
+
+/// Because the `Shape` trait is not dyn safe, it can be difficult to store
+/// Collections of `Shape` items in hetereogenuous collections.
+///
+/// It allows an external `Shape` impl to be provided as an extension point
+/// for shape impls provided by external crates.  This defaults to an
+/// uninhabited type.
+#[derive(Clone, Debug, PartialEq)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[non_exhaustive]
+pub enum PrimitiveShape<External = _never_shape::NeverShape>
```

Comment:
Could this be something like ConcreteShape (suggestion, not recommendation)?

Reply:
I think it is probably the best name yet, I'll hold off on committing the change a day or so in case any other renaming ideas come in though

Reply:
I let this slip through the cracks, but did this rename, so resolving.